pkg/golinters/gocritic.go

```diff
@@ -31,13 +32,13 @@ var (
 	isGoCriticDebug = logutils.HaveDebugTag(logutils.DebugKeyGoCritic)
 )
 
-func NewGoCritic(settings *config.GoCriticSettings, cfg *config.Config) *goanalysis.Linter {
+func NewGoCritic(settings *config.GoCriticSettings, lintConfigDirGetter func() string) *goanalysis.Linter {
```

Comment:
not sure about this change: a struct seems better than a closure.

Reply:
history of change:

1. confused by ambiguous API – what fields of config are needed by GoCritic oO?
2. changed to passing `lintConfigDir` only
3. realized that `Config.cfgDir` filled after `Manager.GetAllSupportedLinterConfigs` call
4. changed to factory

do you think it's better to return it back?

Reply:
yes, I prefer the usage of `cfg` because it's more homogenous with the other linters when they need to use the Config.

Reply:
I implemented an option between - keep `cfg` in constructor signature, but narrow usage of its inside
(more obvious for newcomers)